istioctl/cmd/testdata/auth/authz-policy.golden

```diff
@@ -1,16 +1,4 @@
 apiVersion: rbac.istio.io/v1alpha1
-kind: ServiceRole
```

Comment:
Was there a reason for this change?

Reply:
Yes. I documented the reason here https://github.com/istio/istio/pull/13960/files#diff-9126826617bbc3a02bc93064ea5a1c0eR77 

Per @yangminzhu's request, I will make it more flexible but at the cost of deep copies.